x-pack/plugins/secops/public/store/local/timeline/helpers.ts

```diff
@@ -187,20 +187,50 @@ export const applyDeltaToCurrentWidth = ({
   };
 };
 
-interface AddTimelineProviderParams {
-  id: string;
-  provider: DataProvider;
-  timelineById: TimelineById;
-}
+const addAndToProviderInTimeline = (
+  id: string,
+  provider: DataProvider,
+  timeline: TimelineModel,
+  timelineById: TimelineById
+): TimelineById => {
+  const alreadyExistsProviderIndex = timeline.dataProviders.findIndex(
+    p => p.id === timeline.highlightedDropAndProviderId
+  );
+  const newProvider = timeline.dataProviders[alreadyExistsProviderIndex];
+  const alreadyExistsAndProviderIndex = newProvider.and.findIndex(p => p.id === provider.id);
```

Comment:
I don't think there's a potential runtime exception here, but just to confirm: [Array.prototype.findIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) invoked on [line 196 above](https://github.com/elastic/kibana/pull/29101/files#diff-c09ec5a847684b15bf7a17ad3d2a7142R196) can return `-1`, but we never expect it to, correct?

I'm asking because if `alreadyExistsProviderIndex` evaluates to `-1`, will the result of evaluating the following expression on [line 199](https://github.com/elastic/kibana/pull/29101/files#diff-c09ec5a847684b15bf7a17ad3d2a7142R199)

```
const newProvider = timeline.dataProviders[alreadyExistsProviderIndex];
``` 
would evaluate to an invalid `newProvider`, which in-turn would cause a runtime exception on this line when `newProvider.and.findIndex()` is applied (on this line).

Reply:
you are correct, it is not possible to get an error here because highlightedDropAndProviderId can only come from a dataProvider on the first level and if it is blank you just add a normal data provider not an and data provider.